Libs/Gst/Structs/Message.cs

```diff
@@ -0,0 +1,141 @@
+﻿using System;
+using System.Runtime.InteropServices;
+
+namespace Gst
+{
+    public class MessageTypeMismatchException : Exception
+    {
+        public MessageTypeMismatchException(MessageType actual, MessageType expected)
+            : base($"Expected message type of {expected.ToString()} but received ${actual.ToString()}.")
+        {
+            
+        }
+    }
+    
+    public partial struct Message
+    {
+        public MessageType Type => type;
+
+        public Structure GetStructure()
+        {
+            // Marshal this structure
+            IntPtr thisPtr = Marshal.AllocHGlobal(Marshal.SizeOf(this));
+            Marshal.StructureToPtr(this, thisPtr, false);
+            
+            // Native Call
```

Comment:
I think we should avoid redundant comments. It is obvious that the following call makes a native call. The same goes for the rest of the comments of this method (except the "TODO" one). Comments should add additional value to the code which is not obvious for the reader. Even in case we need such a comment we should probably wonder why the code is not selfexplanatory enough. And if we think that we need the comment after this consideration we should keep it.

We can keep them for now but I would like to get rid of them bit by bit.

Reply:
This is a leftover from when I was trying to fix memory corruption issues with GstMessage. I had comments and code everywhere, so it was just indicating that 'x' section was for native calling, 'y' was for marshalling, etc.

We can get rid of it.

Reply:
Fixed in latest commit